postcss/theme.js

```diff
@@ -136,6 +140,8 @@ module.exports = (options) => {
                 const theme = supportedThemes.find(t => atRule.params.includes(t));
                 if (!theme) throw new Error(`Encountered unsupported theme ${atRule.params}. Allowed themes are ${supportedThemes.join(', ')}`);
 
+                atRule.parent.hasThemeAtRule = true;
```

Comment:
Is there a way we can set this on `root` instead? Because I worry that if the rule is deeply nested this means we won't run the plugin:

```scss
.foo {
     .bar {
          .baz {
                @theme (dark) {
                      background: red;
                 }
          }
     }
}
```

That said it might be fine, because I think only the compiled scss will get passed to the plugin.

Reply:
Removed this logic in favor of leaning on the reset of `rules`.